homeassistant/components/google/calendar.py

```diff
@@ -147,18 +140,30 @@ async def async_get_events(self, hass, start_date, end_date):
         params["timeMin"] = start_date.isoformat("T")
         params["timeMax"] = end_date.isoformat("T")
 
+        event_list = []
         events = await hass.async_add_executor_job(service.events)
+        page_token = None
+        while True:
+            page_token = await self.async_get_events_page(
+                hass, events, params, page_token, event_list
+            )
+            if not page_token:
+                break
+        return event_list
+
+    async def async_get_events_page(self, hass, events, params, page_token, event_list):
```

Comment:
Make it a sync function that tracks the token and makes more requests if necessary. That's more efficient than doing that check in async in between calls.

Reply:
Are you sure? It would also imply calling the Google ``list`` API synchronously, which is a behavior change in comparison to the original code.